usrsctplib/user_socketvar.h

```diff
@@ -403,15 +403,18 @@ void	sofree(struct socket *so);
 
 #define	soref(so) do {							\
 	SOCK_LOCK_ASSERT(so);						\
+	KASSERT((so)->so_count >= 0, ("soref"));			\
 	++(so)->so_count;						\
 } while (0)
 
 #define	sorele(so) do {							\
 	ACCEPT_LOCK_ASSERT();						\
 	SOCK_LOCK_ASSERT(so);						\
 	KASSERT((so)->so_count > 0, ("sorele"));			\
-	if (--(so)->so_count == 0)					\
+	if (--(so)->so_count == 0) {					\
+		(so)->so_count = -1;					\
```

Comment:
If we actually do this, `sofree()` will return early, since `so->so_count != 0` would be true.

Reply:
Yes, you're right; see my follow-up commit.

Reply:
I've rebased it.